File: app/src/main/java/jp/searchwakayamatoilet/AboutAppListItem.java
package jp.searchwakayamatoilet;

import android.support.annotation.NonNull;

import android.databinding.BaseObservable;
import android.databinding.Bindable;

/**
 * Created by masanori on 2016/07/31.
 * List view item class.
 */
public class AboutAppListItem extends BaseObservable{
    private String areaTitle = "";
    private boolean hasAreaTitle = false;
    private String itemTitle = "";
    private boolean hasItemTitle = false;
    private String description = "";
    private String link = "";

    public AboutAppListItem(@NonNull String newAreaTitle, boolean newHasAreaTitle, @NonNull String newItemTitle
            , boolean newHasItemTitle, @NonNull String newDescription, @NonNull String newLink){
        areaTitle = newAreaTitle;
        hasAreaTitle = newHasAreaTitle;
        itemTitle = newItemTitle;
        hasItemTitle = newHasItemTitle;
        description = newDescription;
        link = newLink;
    }
    @Bindable
    public String getAreaTitle(){
        return areaTitle;
    }
    @Bindable
    public boolean getHasAreaTitle(){
        return hasAreaTitle;
    }
    @Bindable
    public String getItemTitle(){
        return itemTitle;
    }
    @Bindable
    public boolean getHasItemTitle(){
        return hasItemTitle;
    }
    @Bindable
    public String getDescription(){
        return description;
    }
    @Bindable
    public String getLink(){
        return link;
    }
}

File: app/src/main/java/jp/searchwakayamatoilet/AboutAppListItem.kt
package jp.searchwakayamatoilet

import android.databinding.BaseObservable
import android.databinding.Bindable

/**
 * Created by masanori on 2016/06/21.
 */
class AboutAppListItem(newAreaTitle: String, newHasAreaTitle: Boolean
                       , newItemTitle: String, newHasItemTitle: Boolean, newDescription: String, newLink: String): BaseObservable(){
    private var areaTitle = ""
    private var hasAreaTitle = false
    private var itemTitle = ""
    private var hasItemTitle = false
    private var description = ""
    private var link = ""

    init{
        areaTitle = newAreaTitle
        hasAreaTitle = newHasAreaTitle
        itemTitle = newItemTitle
        hasItemTitle = newHasItemTitle
        description = newDescription
        link = newLink
    }
    @Bindable fun getAreaTitle(): String{
        return areaTitle
    }
    @Bindable fun getHasAreaTitle(): Boolean{
        return hasAreaTitle
    }
    @Bindable fun getItemTitle(): String{
        return itemTitle
    }
    @Bindable fun getHasItemTitle(): Boolean{
        return hasItemTitle
    }
    @Bindable fun getDescription(): String{
        return description
    }
    @Bindable fun getLink(): String{
        return link
    }
}